### DISCUSSION TITLE
Include openapi callback spec for operation/route

### DISCUSSION THREAD

**Discussion by booooh** [fastapi/fastapi#7979] [Questions]
**Description**

It's my understanding that OpenAPI 3.0 added support for callback objects (https://swagger.io/docs/specification/callbacks/)

As such, it allows defining the schema/model for the operation that the callback would need to implement.

How can I Include the details of a callback specification for an operation?
I couldn't find a way to attach this information to a Route/Operation.

Ideally, I would be able to define the schema of the request that would be handled by the callback.

**Additional context**
This is the spec definition for callback objects:
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#callbackObject

**Comment by booooh** [comment]
If anyone has some information on how to add these, that would be great!
Also, if this isn't directly supported, I'd be happy to try to put a PR together, but would need some pointers on what I should be looking at.

Thanks!

**Comment by booooh** [comment]
@tiangolo  I've started a  PR with a basic implementation that enables creating APIRoutes for callback objects. If you can take a look to see if this is along the lines of what you'd expect for this -- that would be great.

**Comment by tiangolo** [comment]
Thanks for your PR @booooh ! As it's already merged, I'm gonna close this issue now.
